.github/workflows/build-package-docs.yaml

```
@@ -120,3 +131,88 @@ jobs:
         name: package-docs-build
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
+
+  deploy-package-docs:
+    if: fromJSON(github.event.inputs.deploy)
+    needs: build-package-docs
+    runs-on: ubuntu-latest
+    environment:
+      name: deploy-package-docs
+      url: https://github.com/ansible/ansible-documentation
+    env:
+      TARGET: ${{ github.event.inputs.deployment-environment }}
+      DEST_REPO: ansible-community/package-doc-builds
+      USER_EMAIL: "41898282+github-actions[bot]@users.noreply.github.com"
+      USER_NAME: "github-actions[bot]"
+    steps:
+    - name: Download the build archive
+      uses: actions/download-artifact@v4
+      with:
+        name: package-docs-build
+
+    - name: Extract the tarball
+      run: >-
+        tar -xvzf
+        ansible-package-docs-html-*.tar.gz
+        --one-top-level
+
+    - name: Set the production branch
+      if: env.TARGET == 'production'
+      run: >-
+        echo "BRANCH=${{
+          github.event.inputs.repository-branch
+        }}" >> "${GITHUB_ENV}"
```

Better use an intermediate env variable:
```suggestion
      env:
        BRANCH_NAME: ${{ github.event.inputs.repository-branch }}
      run: >-
        echo "BRANCH=${BRANCH_NAME}" >> "${GITHUB_ENV}"
```
Otherwise you can run into all kind of quoting issues.

---

Thanks @felixfontein